Assets/Scripts/Core/Puzzles/PuzzleController.cs

```diff
@@ -0,0 +1,188 @@
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace OperationBlackwell.Core {
+	public class PuzzleController : MonoBehaviour {
+		[Header("Puzzles")]
+		[SerializeField] private Puzzle[] puzzles;
+
+		private PuzzleBlock emptyPuzzleBlock_;
+		private PuzzleBlock[,] puzzleBlocks_;
+		private int blocksPerLine_;
+		private enum PuzzleState {
+			Inactive,
+			Active,
+			Solved
+		}
+		private PuzzleState currentState_;
+		private Transform puzzleVictory_;
+
+		[Header("Sizes")]
+		[SerializeField] private float puzzleSize_;
+		[SerializeField] private GameObject puzzleBlockPrefab_;
+		private Vector2 puzzleOffset_;
+
+		private Queue<PuzzleBlock> input_;
+		private bool blockIsMoving_;
+
+		[Header("Shuffle")]
+		[SerializeField] private int shuffleCount_;
+		private int shuffleCountRemaining_;
+		Vector2Int prevShuffle_;
+
+		[Header("Move Timings")]
+		[SerializeField][Range(0, 1)] private float shuffleMoveTime_;
+		[SerializeField][Range(0, 1)] private float defaultMoveTime_;
+
+		private void Start() {
+			foreach(Transform child in transform) {
+				puzzleVictory_ = child;
+			}
+			puzzleVictory_.gameObject.SetActive(false);
+			CreatePuzzle(0);
+		}
+
+		public void CreatePuzzle(int puzzleIndex) {
+			blocksPerLine_ = puzzles[puzzleIndex].BlocksPerLine();
+			puzzleOffset_ = new Vector2(
+				puzzleSize_ / 2,
+				puzzleSize_ / 2
+			);
+			puzzleBlocks_ = new PuzzleBlock[blocksPerLine_, blocksPerLine_];
+			Texture2D[,] puzzleSlices = ImageSplicer.SplitImage(
+				Utils.TextureFromSprite(puzzles[puzzleIndex].GetSprite()),
+				blocksPerLine_
+			);
+			for(int x = 0; x < blocksPerLine_; x++) {
+				for(int y = 0; y < blocksPerLine_; y++) {
+					GameObject puzzleBlock = Instantiate(puzzleBlockPrefab_, new Vector3(0, 0, 0), Quaternion.identity);
+					puzzleBlock.transform.position = -Vector2.one * (blocksPerLine_ - 1) * .5f +
+						new Vector2(x * puzzleSize_ / (blocksPerLine_ - 1), y * puzzleSize_ / (blocksPerLine_ - 1)) - puzzleOffset_;
+					puzzleBlock.transform.position = new Vector3(puzzleBlock.transform.position.x, puzzleBlock.transform.position.y, -1);
```

Comment:
Coding style violation, missing a space between `-` and `1`.

Reply:
There must not be a space between them as that represents the statement `0 - 1` that becomes `-1`.

Reply:
Oh damn, missed the comma there. My bad.